### ISSUE TITLE
Consider tweaks to Git Guide?

### ISSUE THREAD

**Issue by pizthewiz** [cinder/Cinder#1214]
I'd like to propose two changes to the Git Guide that I thought I'd first gauge interest and agreement on.
- The [Cinder + Git](https://libcinder.org/docs/branch/master/guides/git/index.html) guide suggests one name their Cinder clone `cinder_master` but doesn't rename the `Cinder-OpenCV` block. Could we instead just leave the Cinder clone as is, and leave the resulting folder named `Cinder`? This is especially strange since the folder is named after a branch which is mutable.
- The clone commands use the `git` protocol which in turn uses SSH and requires a GitHub account. Would users reading and needing this guide have a GitHub account? I would think that defaulting to `https` would be the most inclusive, especially for new users.

---

```sh
git clone --recursive git://github.com/cinder/Cinder.git cinder_master
```

```sh
git clone git://github.com/cinder/Cinder-OpenCV.git
```

**Comment by sansumbrella** [comment]
Seems like a nice move, two more reasons to support it:

1) The clone command is simpler when omitting a directory name.
2) github is recommending https by default as of (recently).

**Comment by richardeakin** [comment]
I don't see any reason why we recommend a name for the cloned branch - users can do that if they like, and if they have any (at all) experience with git.

I don't want to recommend using HTTPS protocol because it means you need to write your username and password whenever you use the command line. This isn't a problem for those who use git guis because they store the user account, but that is a small percentage (of which none of the core developers are a part of). To me ssh is the correct one and if you don't have ssh keys installed, you should go do that. If you can't figure out how to install ssh keys, well then you're hopeless. :)

**Comment by paulhoux** [comment]
I have used ```https``` until very recently. Not having to type your password all the time is nice. But it's true that GitHub recommends using ```https``` and I found that using ```git``` on a computer that is used by more than one person makes it impossible to commit under your own name. Setting up SSH is also quite a pain that I would like to avoid for people new to Cinder and Git.

**Comment by richardeakin** [comment]
@paulhoux  you use tortoisegit, right? This is why you don't have to enter your password as it stores it for you. At the command line you need to enter it each time you want to pull, push or do anything that contacts the remote.

Honestly though, this is a C++ framework. We shouldn't even have to spell out the command to do a clone for users, just point them to the repo and that should be enough IMO.

**Comment by braitsch** [comment]
I agree with @richardeakin. It seems reasonable to me to expect that anyone advanced enough to be dabbling in Cinder should be comfortable working with the basics of Git and setting up SSH keys. 

Regardless though, [perhaps it'd be good to include [a link to the setup guide](https://help.github.com/articles/generating-ssh-keys/) on Github?

**Comment by paulhoux** [comment]
@richardeakin : yeah, I prefer using TortoiseGit when possible, because it makes my life a lot easier if I don't have to remember all the git commands. But TortoiseGit does not store passwords out of the box, so when using ```https``` I always have to enter username and password every time I commit/push (or pull from a private repo). 

Now that I've setup the SSH keys on my system, I prefer using ```git```, of course. But setting up the keys took me an hour or so, and forced me to reinstall both Git and TortoiseGit because I needed different setup options. It was a hassle, and will be again when I move to a different system. Not insurmountable, but it might put off new Cinder users. These users may not have a lot of experience with versioning systems, or may have been using SVN or Mercurial or what have you. We can't just assume everyone knows how to setup and use Git.

While Cinder is intended for advanced, serious C++ programmers, it would be a mistake to adopt an elitist attitude.

**Comment by richardeakin** [comment]
Huh, an hour.. for me it usually takes < 30 seconds to setup git on a new machine. But I use a terminal. :) 

If you use Github for Windows (or Mac) it basically does everything for you, including installing the ssh key to your account.

@braitsch I think that's a good idea to include that link in the setup guide, it is the first thing that comes up when you google "github install ssh key", but it can't hurt to tell people 'this is the way you do it'.
